src/dt2js.js

```diff
@@ -1,21 +1,60 @@
 'use strict'
 
-var yaml = require('js-yaml')
+var yap = require('yaml-ast-parser')
 var dtexp = require('datatype-expansion')
 var constants = require('./constants')
 var utils = require('./utils')
-
+var fs = require('fs')
+var path = require('path')
+var deep = require('deep-get-set')
+deep.p = true
 /**
  * Get RAML Data Types context.
  *
  * @param  {string} ramlData - RAML file content.
  * @returns  {Object} - RAML data types context.
  */
-function getRAMLContext (ramlData) {
-  var yamlContent = yaml.safeLoad(ramlData)
-  return yamlContent.types
+function getRAMLContext (ramlData, rootFileDir) {
+  rootFileDir = rootFileDir || '.'
+  var ast = yap.load(ramlData)
+  var jsContent = {}
+  traverse(jsContent, ast, rootFileDir)
+  return jsContent.types
 }
 
+function traverse (obj, ast, rootFileDir) {
+  function recurse (keys, currentNode) {
+    if (currentNode.key) {
+      keys = keys.concat([currentNode.key.value])
+    }
+    if (currentNode.value && currentNode.value.kind === 5) {
```

Comment:
it would be nice to have a comment before each of the conditions of this block to understand what the conditions `currentNode.value && currentNode.value.kind === 5` or `currentNode.value && currentNode.value.value` for example mean.